urld44fdf4fee_nadilsons_reflection_dsl_tgz-pJ8-br_com_bit_ideias_reflection_SimplePerformanceTestJ8/src/br/com/bit/ideias/reflection/rql/query/Rql.java

```diff
@@ -0,0 +1,258 @@
+        while(matcher.find()) {
+            connectors.add(QueryConnector.valueOf(matcher.group().trim().toUpperCase()));
+        }
+            
+        String[] parts = OPERATORS_PATTERN.split(partial);
+        
+        for (String part : parts) {
+            if(part.trim().length() == 0)
+                continue;
+            complexPart.addPart(new ExpressionPart(part));
+        }
+        
+        for (QueryConnector connector : connectors) {
+            complexPart.addConnector(connector);
+        }
+    }
+
+    protected static Expression parseSimple(String rql) {
+        String originalRql = rql;
+        rql = rql.trim();
+        String lowerRql = rql.toLowerCase();
+        if(!EXPRESSION_PATTERN.matcher(lowerRql).matches())
+            throw new SyntaxException(String.format("There is an error on this part of your rql => %s", rql));
+        
+        //now we get what it is demmanded
+        int pos = rql.indexOf(' ');
+        String clauseName = rql.substring(0, pos).toUpperCase().trim();
+        
+        QueryClause clause = QueryClause.valueOf(clauseName);
+        
+        rql = rql.substring(pos).trim();
+        pos = rql.indexOf(' ');
+        String op = rql.substring(0, pos).toUpperCase().trim();
+        
+        QueryOperator operator = transformToQueryOperator(op);
+        
+        if(!clause.acceptsOperator(operator))
+            throw new SyntaxException(String.format("%s does not work with %s => %s", clauseName, op, rql));
+        
+        String rightHand = rql.substring(op.length()).trim();
+        
+        if(rightHand.startsWith("'") || rightHand.startsWith("(")) {
+            return operator.getExpression(clause, removeEdges(rightHand));
+        } else {
+            throw new SyntaxException(String.format("There is an error on the right hand operand of this query => %s", originalRql));
+        }
+    }
+
+    private static String removeEdges(String text) {
+        return text.substring(1, text.length() - 1);
+    }
+
+    private static QueryOperator transformToQueryOperator(String op) {
+        QueryOperator operator = null;
+        if("=".equals(op))
+            operator = QueryOperator.EQ;
+        else if("!=".equals(op))
+            operator = QueryOperator.NE;
+        else
+            operator = QueryOperator.valueOf(op);
+        return operator;
+    }
+
+    private void validateRestrictions(String rql) {
+        char chr = '-';
+        Stack<Integer> stack = new Stack<Integer>();
+        Integer integer = Integer.valueOf(0);
+        boolean isString = false;
+        
+        for (int i = 0; i < rql.length(); i++) {
+            chr = rql.charAt(i);
+            
+            switch(chr) {
+                case '(':
+                    if(!isString) stack.add(integer);
+                break;
+                case ')':
+                    try {
+                        if(!isString) stack.pop();
+                    } catch (EmptyStackException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)